docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+
+The Dojo build process allows for imports to be conditionally removed through the use of _has pragmas_. These pragmas are simply strings that refer to a specific feature. If the feature is statically set at build time then the build system will evaluate the pragma and if `true` the next import is removed.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"features": {
+			"foo": true,
+			"bar": false
+		}
+	}
+}
+```
+
+> main.ts
+
+```ts
+"has('foo')";
+import 'a'; // removed
+'!has("bar")';
+import 'b'; // removed
+'!has("foo")';
+import 'c';
+'has("baz")';
+import 'd';
+```
+
+In the above example, imports `a` and `b` would be removed because their values evaluate to `true`. Import `c` will remain because it evaluates to `false`. Import `d` will remain because the `baz` feature has not been statically defined.
+
+## Known features
+
+The following features are recognized by the Dojo system and may be used internally to optimize code or shim features. If multiple features are specified then the intersection of available features will be returned.
+
+-   android
+-   chrome
+-   edge
+-   firefox
+-   ie11
+-   ios
+-   node
+-   node8
+-   safari
+
+## Provided Features
+
+These features are provided by the build system to help identify a specific environment or mode of operation.
+
+| Feature Flag        | Description                                                                                                                                                                                                                                              |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `debug`             | Provides a way to create a code path for code that is only usable when debugging or providing enhanced diagnostics that are not desired in a _production_ build. Defaults to `true` but should be configured statically as `false` in production builds. |
+| `host-browser`      | Determines if the current environment contains a `window` and `document` object in the global context, therefore it is generally safe to assume the code is running in a browser environment.                                                            |
+| `host-node`         | Attempts to detect if the environment appears to be a node environment.                                                                                                                                                                                  |
+| `build-time-render` | Statically defined by the build-time rendering system during build-time rendering.                                                                                                                                                                       |
+
+# Build-time Rendering
```

Comment:
This is a flagship feature along with the PWA support, we should move this to the top.